src/app/context/appState.tsx

```diff
@@ -0,0 +1,121 @@
+import produce, { Draft } from 'immer';
+import {
+  createContext,
+  Dispatch,
+  ReactNode,
+  useContext,
+  useReducer,
+} from 'react';
+
+import {
+  DataState,
+  Measurements,
+  mergeMeasurements,
+  getEmptyDataState,
+} from '../../data/DataState';
+import { assertUnreachable } from '../../utils/assert';
+
+import { getFirstMeasurement, getMeasurementOrFail } from './data.helpers';
+
+export interface AppState {
+  data: DataState;
+  isLoading: boolean;
+  view: {
+    selectedMeasurement?: string;
+  };
```

Comment:
It's fine as a first iteration, but I think that the goal is to have one selected measurement per "tab" (i.e. if you select an IR spectrum X, then go to the Raman tab and select Raman spectrum Y, then go back to the IR tab, it should remember that spectrum X is selected there.